flows/engine/engine.go

```diff
@@ -143,24 +139,30 @@ func resumeNode(run flows.FlowRun, node flows.Node, step flows.Step, event flows
 		return noDestination, step, nil
 	}
 
-	err := wait.End(run, step, event)
+	// try to resume our wait with the first caller event
+	err := wait.End(run, step, callerEvents[0])
```

Comment:
Maybe waits should take a list of events? This seems like a very specific and subtle contract with our callers.